app/views/feeds/_form_apple_config.html.erb

```diff
@@ -3,28 +3,28 @@
     <div class="card-header card-header-light d-flex justify-content-between">
       <h2 class="card-title h5 flex-grow-1"><%= t(".title") %></h2>
     </div>
-    <div class="card-body">
+    <div class="card-body" data-controller="apple-key">
       <div class="mb-4">
         <p><%= t(".description") %></p>
         <a href="https://help.prx.org"><%= t(".guide_link") %></a>
       </div>
       <%= form.fields_for :apple_config do |config_form| %>
         <%= config_form.fields_for :key do |key_fields| %>
-          <div class="mb-4">
-            <div class="form-floating">
-              <%= key_fields.text_field :provider_id, disabled: config_form.object.persisted? %>
-              <%= key_fields.label :provider_id %>
-            </div>
+          <div class="mb-4 form-floating">
+            <%= key_fields.text_field :provider_id, disabled: config_form.object.persisted? %>
+            <%= key_fields.label :provider_id, "Provider ID" %>
           </div>
-
-          <div>
-            <div class="form-floating">
-              <%= key_fields.text_field :key_id, disabled: config_form.object.persisted? %>
-              <%= key_fields.label :key_id %>
-            </div>
+          <div class="form-floating input-group">
+            <%= key_fields.text_field :key_id, disabled: config_form.object.persisted?, data: {apple_key_target: "key", action: "keyup->apple-key#convertKeyToB64"} %>
```

Comment:
the `keyup` event trigger is for the case if a user decides to just copy/paste the key into the field, i still want to run the method that converts it to base64 for the (hidden) `key_pem_b64` field.

Reply:
So slight UI update... re-reading a bunch of threads, it _sounds_ like the filename of the pem that Apple gives you already contains the provider_id and key_id?  In which case, we should try to auto parse that for users.

So complete workflow:

1. Click on the New Apple Feed button, form shows _only_ a file-upload field
2. Upload your PEM with a name like `<some-uuid-4-looking-provider-id>_<10-digit-key-id>.p8`
3. Your stimulus controller parses that filename, fills in 2 hidden fields for `provider_id` and `key_id`
4. Your stimulus controller also _reads_ the file, and fills in the `key_pem_b64` hidden field
5. Click save, browser POSTs those 3 hidden fields
6. If any validation failed, error message is displayed on the form under the file-upload field

Then once you already have a saved Feeds::AppleSubscription, the form shows the (current) readonly `provider_id` and `key_id` fields.